Userland/Libraries/LibCore/Object.h

```diff
@@ -181,8 +181,6 @@ class Object
     void increment_inspector_count(Badge<InspectorServerConnection>);
     void decrement_inspector_count(Badge<InspectorServerConnection>);
 
-    virtual bool load_from_json(const JsonObject&, RefPtr<Core::Object> (*)(const String&)) { return false; }
```

Comment:
Wait so, there's definitely things that aren't widgets that get loaded from GML. Like... layouts? were any of the non-widget registrations also Core::Objects? I'm curious why moving this API from Core::Object to GUI::Widget is "fine".

Reply:
Technically, yes, layouts aren't widgets. But no layout specializes the load_from_json, so they don't expect to be loaded with any special parameters etc. anyways. The registrations are still found because we check for subclasses of Layout in the appropriate places instead of Widget. And I can't find any non-widget object except layouts being used in GML, so I don't see the problem.